datafusion/spark/src/function/string/format_string.rs

```diff
@@ -335,7 +335,7 @@ impl<'a> Formatter<'a> {
                 let (spec, rest) = take_conversion_specifier(
                     rest,
                     current_argument_index,
-                    arg_types[current_argument_index - 1].clone(),
+                    &arg_types[current_argument_index - 1].clone(),
```

Comment:
```suggestion
                    &arg_types[current_argument_index - 1],
```

No point in passing by reference if we still clone it at the callsite